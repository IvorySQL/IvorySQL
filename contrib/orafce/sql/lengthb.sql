set datestyle to ISO,YMD;
set search_path to oracle, pg_catalog;
SELECT lengthb(192);
SELECT lengthb(192::int2);
SELECT lengthb(192::int4);
SELECT lengthb(192::int8);
SELECT lengthb(192.922);
SELECT lengthb(192.922::numeric);

SELECT lengthb('192');
SELECT lengthb('Highgo DB!');
SELECT lengthb('今天');
SELECT lengthb('Highgo DB!'::char(10));
SELECT lengthb('Highgo DB!'::char(20));
SELECT lengthb('Highgo DB!'::varchar(20));
SELECT lengthb('Highgo DB!'::varchar2(20));
SELECT lengthb('Highgo DB!'::nvarchar2(20));
SELECT lengthb('Highgo DB!'::text);
SELECT lengthb('今天'::text);
SELECT lengthb('今天是Monday'::char(20));

SELECT lengthb('2019-12-12'::date);
SELECT lengthb('2019-12-12'::timestamp);
SELECT lengthb(null);
SELECT lengthb('');
SELECT lengthb(' ');
SELECT lengthb(0::int);
SELECT lengthb(-2::int);

reset search_path;