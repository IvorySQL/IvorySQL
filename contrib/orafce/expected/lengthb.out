set datestyle to ISO,YMD;
set search_path to oracle, pg_catalog;
SELECT lengthb(192);
 lengthb 
---------
       3
(1 row)

SELECT lengthb(192::int2);
 lengthb 
---------
       3
(1 row)

SELECT lengthb(192::int4);
 lengthb 
---------
       3
(1 row)

SELECT lengthb(192::int8);
 lengthb 
---------
       3
(1 row)

SELECT lengthb(192.922);
 lengthb 
---------
       7
(1 row)

SELECT lengthb(192.922::numeric);
 lengthb 
---------
       7
(1 row)

SELECT lengthb('192');
 lengthb 
---------
       3
(1 row)

SELECT lengthb('Highgo DB!');
 lengthb 
---------
      10
(1 row)

SELECT lengthb('今天');
 lengthb 
---------
       6
(1 row)

SELECT lengthb('Highgo DB!'::char(10));
 lengthb 
---------
      10
(1 row)

SELECT lengthb('Highgo DB!'::char(20));
 lengthb 
---------
      20
(1 row)

SELECT lengthb('Highgo DB!'::varchar(20));
 lengthb 
---------
      10
(1 row)

SELECT lengthb('Highgo DB!'::varchar2(20));
 lengthb 
---------
      10
(1 row)

SELECT lengthb('Highgo DB!'::nvarchar2(20));
 lengthb 
---------
      10
(1 row)

SELECT lengthb('Highgo DB!'::text);
 lengthb 
---------
      10
(1 row)

SELECT lengthb('今天'::text);
 lengthb 
---------
       6
(1 row)

SELECT lengthb('今天是Monday'::char(20));
 lengthb 
---------
      26
(1 row)

SELECT lengthb('2019-12-12'::date);
 lengthb 
---------
      19
(1 row)

SELECT lengthb('2019-12-12'::timestamp);
 lengthb 
---------
      19
(1 row)

SELECT lengthb(null);
 lengthb 
---------
        
(1 row)

SELECT lengthb('');
 lengthb 
---------
       0
(1 row)

SELECT lengthb(' ');
 lengthb 
---------
       1
(1 row)

SELECT lengthb(0::int);
 lengthb 
---------
       1
(1 row)

SELECT lengthb(-2::int);
 lengthb 
---------
       2
(1 row)

reset search_path;
